### PR TITLE
[react-native] Add declaration for FileReader and Blob global type

### PR DESCRIPTION
**Pull request by maltoze**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [https://github.com/facebook/react-native/blob/master/Libraries/Blob/FileReader.js](https://github.com/facebook/react-native/blob/master/Libraries/Blob/FileReader.js)
[https://github.com/facebook/react-native/blob/master/Libraries/Blob/Blob.js](https://github.com/facebook/react-native/blob/master/Libraries/Blob/Blob.js)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by alloy** [comment]
Nice, thanks 🙏

**Comment by danilobuerger** [comment]
@maltoze @alloy This seems to break the typing to be able to upload local files via fetch, for example:
```ts
fetch("url", {
    body: { uri: "path" },
    headers: {
      "Content-Type": "type",
    },
    method: "PUT",
});
```
now leads to 
```
error TS2322: Type '{ uri: string; }' is not assignable to type 'string | ArrayBuffer | DataView | Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | ... 8 more ... | undefined'.
  Object literal may only specify known properties, and 'uri' does not exist in type 'ArrayBuffer | DataView | Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | Uint16Array | ... 5 more ... | FormData'.
```
This was probably never meant to be typed before as `Blob` was an empty interface and just allowed this code.

**Comment by maltoze** [comment]
@danilobuerger I think you should use `JSON.stringify` on `body`.

**Comment by danilobuerger** [comment]
@maltoze no, that doesn't work as rn will not recognize the upload uri if its a string. Also see the flow typing:
https://github.com/facebook/react-native/blob/master/Libraries/Network/convertRequestBody.js#L22

**Comment by alloy** [comment]
@maltoze Will you send a PR to update this?

**Comment by maltoze** [comment]
@danilobuerger Sorry, I don't understand when to use `body: { uri: "path" }`. 
But maybe we can solve this problem by adding the type `{uri: string}` to `BobyInit_`. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-native/globals.d.ts#L81
@alloy Am I right?

**Comment by danilobuerger** [comment]
@maltoze its used to upload a file at the specified path.

**Comment by kuasha420** [comment]
Also hit the same error and I have opened a PR which makes the BodyInit_ more or less compatible with the flow type. 

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50452

File: types/react-native/globals.d.ts [hunk 1]
```diff
@@ -35,7 +35,21 @@ declare interface WindowOrWorkerGlobalScope {
...
+}
+
+interface BlobOptions {
+    type: string;
+    lastModified: number;
+}
+
+declare var Blob: {
+    prototype: Blob;
+    new (blobParts?: Array<Blob | string>[], options?: BlobOptions): Blob;
```
Comment (briice) [inline types/react-native/globals.d.ts:51]: I think there might be an error here, as this constructor expects an array of `String` or `Blob`, not an array of arrays, right? -> [https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob)

```
   new (blobParts?: (Blob | string)[], options?: BlobOptions): Blob;
```

Comment (maltoze) [reply types/react-native/globals.d.ts:51]: yes...I will send a PR to fix this.

Comment (solimant) [reply types/react-native/globals.d.ts:51]: @maltoze - I think `options` being of type `BlobOptions` puts it a little off from what [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob) and the [File API](https://w3c.github.io/FileAPI/#blob-section) specifies. Particularly, `lastModified` doesn't seem to be part of the specifications, which only call out `type` and `endings`.

Comment (maltoze) [reply types/react-native/globals.d.ts:51]: https://github.com/facebook/react-native/blob/2bf866e4018ea72c1f1c92c806db85378c801fb7/Libraries/Blob/BlobTypes.js#L26

**Pull request merged**